### PR TITLE
Add Stream JSON parser

### PR DESCRIPTION
**Pull request by longsihua2003**
## Summary by CodeRabbit

- **New Features**
	- Introduced the `AIChat` class for managing multi-conversation scenarios with enhanced memory capabilities.
	- Added support for streaming responses in the `pne.chat()` function, improving real-time interactions.
	- Implemented a new utility for handling and fixing JSON data, enhancing robustness in parsing.

- **Documentation**
	- Expanded documentation to clarify usage of `pne.chat()` with various LLMs and API integrations, including detailed examples.

- **Bug Fixes**
	- Enhanced logic for handling streaming and output schema parameters to improve flexibility in response processing.

### CODE REVIEW

File: tests/test_chat.py [hunk 1]
```diff
@@ -68,10 +68,6 @@ def test_init():
         chat("hello")
         assert str(e.value) == "model or custom_llm must be provided."
 
-    # stream and output_schema and not exist at the same time.
-    with pytest.raises(ValueError):
```
Comment (Undertone0809) [inline tests/test_chat.py:72]: Need to add a test to assert output schema + stream.

File: promptulate/utils/jsonFix.py
Comment (Undertone0809) [inline promptulate/utils/jsonFix.py:1]: json_fix.py

File: example/chat_usage.ipynb
Comment (Undertone0809) [inline example/chat_usage.ipynb:1]: Can you give a screenshot to show your changes?

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

File: example/chat_usage.ipynb
Comment (longsihua2003) [reply example/chat_usage.ipynb:1]: 截屏2024-07-31 19 29 18
截屏2024-07-31 19 29 25

**Review by Undertone0809 — APPROVED** [review]

**Pull request merged**